src/NuGet.Clients/NuGetConsole.Host.PowerShell/RunspaceManager.cs

```diff
@@ -35,12 +37,17 @@ internal class RunspaceManager : IRunspaceManager
             Justification = "We can't dispose it if we want to return it.")]
         private static Tuple<RunspaceDispatcher, NuGetPSHost> CreateAndSetupRunspace(IConsole console, string hostName)
         {
-            Tuple<RunspaceDispatcher, NuGetPSHost> runspace = CreateRunspace(console, hostName);
-            SetupExecutionPolicy(runspace.Item1);
-            LoadModules(runspace.Item1);
-            LoadProfilesIntoRunspace(runspace.Item1);
+            return NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
I'm not sure this JTF.Run achieves much.

If `CreateRunspace` needs the UI thread, there's no call to switch to UI thread, which means that this method must have been called on the UI thread. JTF.Run blocks the current thread, allowing only dependent tasks to run, so this doesn't help reduce UI delays.

If the current method was called on a background thread, then calling JTF.Run and awaiting TaskScheduler.Default does just uses 2 threads instead of 1 (blocks one thread, other one does work).

Unless my understanding of JTF is wrong.

Reply:
Problem was  [ SetupExecutionPolicy(runspace.Item1);](https://github.com/NuGet/NuGet.Client/blob/14fa722fbfeb9b485e86dac5c20e589de9138bf5/src/NuGet.Clients/NuGetConsole.Host.PowerShell/RunspaceManager.cs#L39) runs only for background thread. So this is `TaskScheduler.Default` helps with that one.
@srdjanjovcic  can you comment here?